trulens/nn/models/pytorch.py

```diff
@@ -442,26 +477,32 @@ def qoi_bprop(
             as the input.
         """
         B = get_backend()
-        if attribution_cut is None:
-            attribution_cut = InputCut()
-        if to_cut is None:
-            to_cut = OutputCut()
+
+        doi_cut, to_cut, attribution_cut = self._qoi_bprop_defaults(
+            doi_cut=doi_cut, to_cut=to_cut, attribution_cut=attribution_cut)
+
         self._model.train()
+
         y, zs = self.fprop(
-            model_args,
-            model_kwargs,
-            doi_cut=doi_cut if doi_cut else InputCut(),
+            model_args=model_args,
+            model_kwargs=model_kwargs,
+            doi_cut=doi_cut,
             to_cut=to_cut,
             attribution_cut=attribution_cut,
             intervention=intervention,
             return_tensor=True)
 
         y = to_cut.access_layer(y)
         grads_list = []
+
         for z in zs:
             z_flat = ModelWrapper._flatten(z)
             qoi_out = qoi(y)
 
+            # TODO(piotrm): this sum is a source of much bugs for me when using
+            # attributions. If one wants a specific QoI, the sum hides the bugs
+            # in the definition of that QoI. It might be better to give an error
+            # when QoI is not a scalar.
```

Comment:
multi qoi is used for things like average qoi. its morre efficient to do it here and have 1 gradient rather than many gradients

Reply:
I agree but I'm saying that even if one wants one gradient, like class qoi, but implements it incorrectly (assuming class qoi was not already provided for them), then this sum might hide their mistake(s).

Reply:
let add a warning that this sum is usually for multi output.

Reply:
Added a warning only if more than 1 dimension is being aggregated.